controllers/complianceeventsapi/auth.go

```diff
@@ -0,0 +1,165 @@
+package complianceeventsapi
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/stolostron/rbac-api-utils/pkg/rbac"
+	"gopkg.in/square/go-jose.v2/jwt"
+	authorizationv1 "k8s.io/api/authorization/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	policyv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+)
+
+func getReadAuth(ctx context.Context, config *rest.Config,
+	token, managedClusterName string, client client.Client,
+) (bool, error) {
+	allManagedClusterRules, err := getAllManagedClusterRules(ctx, config, token, client)
+	if err != nil {
+		return false, err
+	}
+
+	rules, ok := allManagedClusterRules[managedClusterName]
+	if !ok {
+		return false, nil
+	}
+
+	if slices.Contains(rules, "get") || slices.Contains(rules, "*") {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func getAllManagedClusterRules(ctx context.Context, config *rest.Config,
+	token string, client client.Client,
+) (map[string][]string, error) {
+	mangedClusterList := &clusterv1.ManagedClusterList{}
+
+	err := client.List(ctx, mangedClusterList)
```

Comment:
This is not required. It's better to call `GetResourceAccess` without specifying a value for `resourcenames`. Then if the user has access to a subset of clusters, then a list of cluster names is returned. If the user is an admin/has access to all clusters, then `*` is returned. In the case of `*` you don't need to set any filters in the where clause.

Reply:
On second thought, if the user provides a cluster.name or cluster.cluster_id filter, then you'd set `resourcenames` to the filter. Then the returned list will be the subset that they requested.